### PR TITLE
Fix missing 3.17 requests

### PR DESCRIPTION
**Pull request by CGNonofr**

### CODE REVIEW

File: src/language-client-mutualization.ts [hunk 1]
```diff
@@ -201,35 +203,42 @@ function bindClientToServer (
...
-    })
+    })))
+    disposableCollection.push(languageClient.onInlayHintRefresh(bindContext(() => {
+      if (clientCapabilities.workspace?.inlayHint?.refreshSupport ?? false) {
+        clientConnection.sendRequest(InlayHintRefreshRequest.type).catch(error => {
+          options.logger?.error('Unable to send Inlay Hint refresh to client', { error })
+        })
+      }
+    })))
+    disposableCollection.push(languageClient.onInlayHintRefresh(bindContext(() => {
```
Comment (CGNonofr) [inline src/language-client-mutualization.ts]: InlineValue

File: src/language-client-mutualization.ts [hunk 2]
```diff
@@ -201,35 +203,42 @@ function bindClientToServer (
...
+    disposableCollection.push(languageClient.onDiagnosticsRefresh(bindContext(() => {
+      if (clientCapabilities.workspace?.diagnostics?.refreshSupport ?? false) {
         clientConnection.sendRequest(DiagnosticRefreshRequest.type).catch(error => {
           options.logger?.error('Unable to send Diagnostics refresh to client', { error })
         })
       }
-    })
+    })))
+    disposableCollection.push(languageClient.onInlayHintRefresh(bindContext(() => {
+      if (clientCapabilities.workspace?.inlayHint?.refreshSupport ?? false) {
```
Comment (samuel-olivier) [inline src/language-client-mutualization.ts]: shouldn't we move this `if` around the `disposableCollection.push(...)` ? (Same for other `disposableCollection.push`)

File: src/language-client-mutualization.ts (hunk 1 above)
Comment (CGNonofr) [reply src/language-client-mutualization.ts]: af5f191de3285d0383f9efc600637e8f6ba7279e

File: src/language-client-mutualization.ts (hunk 2 above)
Comment (CGNonofr) [reply src/language-client-mutualization.ts]: At the end, this check is useless as the server is already aware if the client supports it or not

=> c1679073c0b04b9d88f388c2ac6ce0002e9c0f74

**Review by samuel-olivier — APPROVED** [review]

**Pull request merged**